tools/refguide_check.py

```diff
@@ -809,6 +852,46 @@ def check_doctests_testfile(fname, verbose, ns=None,
 
     return results
 
+def check_rst(base_path, results, args, dots):
+    if os.path.exists(base_path) and os.path.isfile(base_path):
+        tut_results = check_doctests_testfile(base_path,
+            (args.verbose >= 2), dots=dots,
+            doctest_warnings=args.doctest_warnings)
+        def scratch(): pass        # stub out a "module", see below
+        scratch.__name__ = base_path
+        results.append((scratch, tut_results))
+    for dir_name, subdirs, files in os.walk(base_path, topdown=True):
+        if dir_name in RST_SKIPLIST:
+            if args.verbose > 0:
+                sys.stderr.write('skipping files in %s' % dir_name)
+            files = []
+        for p in RST_SKIPLIST:
+            if p in subdirs:
+                if args.verbose > 0:
+                    sys.stderr.write('skipping %s and subdirs' % p)
+                subdirs.remove(p)
+        test_files = []
+        for f in files:
+            if (os.path.splitext(f)[1] in ('.py', '.rst') and
+                    f not in RST_SKIPLIST):
+                test_files.append(f)
+        if len(test_files) > 1:
+            for fname in test_files:
```

Comment:
Can you extract this to an iterator?

Perhaps
```python
def iter_included_rst_files():
    for dir_name, subdirs, files in os.walk(base_path, topdown=True):
        if dir_name in RST_SKIPLIST:
            if args.verbose > 0:
                sys.stderr.write('skipping files in %s' % dir_name)
            files = []
        for p in RST_SKIPLIST:
            if p in subdirs:
                if args.verbose > 0:
                    sys.stderr.write('skipping %s and subdirs' % p)
                subdirs.remove(p)
        for f in files:
            if (os.path.splitext(f)[1] in ('.py', '.rst') and
                    f not in RST_SKIPLIST):
                yield os.path.join(dir_name, fname)

```

Reply:
done